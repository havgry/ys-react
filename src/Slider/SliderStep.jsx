import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const SliderStep = ({
  children,
  className,
  ...props
}) => {
  const wrapperProps = {
    className: classNames(
      'ys-slider__data-item',
      className,
    ),
    ...props,
  }

  return (
    <span {...wrapperProps}>
      {children}
    </span>
  )
}

SliderStep.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
}

SliderStep.defaultProps = {
  className: '',
}

export default SliderStep
