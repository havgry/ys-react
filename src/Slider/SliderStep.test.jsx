import React from 'react'
import { shallow, mount } from 'enzyme'

import SliderStep from './SliderStep'

describe('<SliderStep>', () => {
  describe('Props', () => {
    it('Throws propType warning if no `children` prop is passed', () => {
      expect(() => (
        shallow(
          <SliderStep />,
        )
      )).toThrow()
    })

    it('Throws propType warning if `children` isn\'t a string', () => {
      expect(() => (
        shallow(
          <SliderStep>
            {100}
          </SliderStep>,
        )
      )).toThrow()
    })

    it('Has correct defaultProps', () => {
      const component = mount(
        <SliderStep>
          Label
        </SliderStep>,
      )

      expect(component.props().className).toBe('')
    })
  })

  describe('Rendering', () => {
    it('Outputs elements with the correct class names', () => {
      expect(
        shallow(
          <SliderStep>
            Step label
          </SliderStep>,
        )
          .hasClass('ys-slider__data-item'),
      ).toBe(true)
    })

    it('Custom `className` is being applied to the container', () => {
      expect(
        shallow(
          <SliderStep className="caracal">
            Step label
          </SliderStep>,
        )
          .hasClass('caracal'),
      ).toBe(true)
    })

    it('Places `children` inside label container', () => {
      expect(
        shallow(
          <SliderStep>
            Label
          </SliderStep>,
        )
          .text(),
      ).toBe('Label')
    })
  })

  describe('Snapshots', () => {
    it('Default', () => {
      expect(
        shallow(
          <SliderStep>
            Step label
          </SliderStep>,
        ),
      ).toMatchSnapshot()
    })

    it('With custom props', () => {
      expect(
        shallow(
          <SliderStep
            className="wild-cats"
            id="lynx"
          >
            Step label
          </SliderStep>,
        ),
      ).toMatchSnapshot()
    })
  })
})
