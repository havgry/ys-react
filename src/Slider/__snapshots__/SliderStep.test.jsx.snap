// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SliderStep> Snapshots Default 1`] = `
<span
  className="ys-slider__data-item"
>
  Step label
</span>
`;

exports[`<SliderStep> Snapshots With custom props 1`] = `
<span
  className="ys-slider__data-item wild-cats"
  id="lynx"
>
  Step label
</span>
`;
